components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1961,6 +2041,11 @@ where
                 assert_eq!(peer_id, self.fsm.peer.peer_id());
                 if !merge_from_snapshot {
                     self.destroy_peer(false);
+                    if let Some(UnsafeRecoveryState::Destroy(_)) =
```

Comment:
Unnecessary. When the peer is destroyed, the `unsafe_recovery_state` is dropped with `self.fsm.peer`